src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -557,12 +557,16 @@ def reflect_account_groups_on_workspace(self):
         tasks = []
         account_groups_in_account = self._account_groups_in_account()
         account_groups_in_workspace = self._account_groups_in_workspace()
+        workspace_groups_in_workspace = self._workspace_groups_in_workspace()
         groups_to_migrate = self.get_migration_state().groups
         logger.info(f"Starting to reflect {len(groups_to_migrate)} account groups into workspace for migration...")
         for migrated_group in groups_to_migrate:
             if migrated_group.name_in_account in account_groups_in_workspace:
                 logger.info(f"Skipping {migrated_group.name_in_account}: already in workspace")
                 continue
+            if migrated_group.name_in_account in workspace_groups_in_workspace:
+                logger.error(f"Skipping {migrated_group.name_in_account}: group already exists in workspace")
```

Comment:
I am thinking about the wording here: "already" makes sense from the integration test perspective, but not from the group migration perspective as it should have been renamed to a temporary name before. So maybe "still" is better

Reply:
Aside from that, we're not really skipping it: this is an error and should cause failure of the group migration. Logging an error will implicitly fail the enclosing workflow (due to log scanning), but maybe these should be gathered and reported as part of the `ManyError` that can be raised at the end.